Userland/Libraries/LibAudio/MP3Loader.cpp

```diff
@@ -144,7 +228,682 @@ bool MP3LoaderPlugin::synchronize()
             m_bitstream->align_to_byte_boundary();
         }
     }
-    return one_counter == 12;
+    if (one_counter != 12)
+        return LoaderError { LoaderError::Category::Format, "Failed to synchronize" };
+    return {};
+}
+
+ErrorOr<MP3::MP3Frame, LoaderError> MP3LoaderPlugin::read_next_frame()
+{
+    while (!m_bitstream->has_any_error()) {
+        TRY(synchronize());
+        MP3::Header header = TRY(read_header());
+        if (header.id != 1 || header.layer != 3) {
+            continue;
+        }
+
+        return read_frame_data(header, m_is_first_frame);
+    }
+    return LoaderError { LoaderError::Category::Internal, "Could not find another frame." };
+}
+
+ErrorOr<MP3::MP3Frame, LoaderError> MP3LoaderPlugin::read_frame_data(MP3::Header const& header, bool is_first_frame)
+{
+    MP3::MP3Frame frame = {};
```

Comment:
At first I thought I wouldn't need it, but it's just a struct so it can be uninitialized. And boy oh boy does it make for a bug hunt if all you have is your ears. Now that it contains a FixedArray, I had to create a constructor for it, so there we go.